lib/masamune/abstract_syntax_tree/prism/node_extensions.rb

```diff
@@ -0,0 +1,74 @@
+module Prism
+  class Node
+    def comment? = false
+
+    def self.node_predicate(name, body = -> { true })
+      define_method(name, &body)
+      Node.define_method(name) { false } unless Node.respond_to?(name, false)
+    end
+
+    def line_number
+      token_location.start_line
+    end
+
+    # #location provides helpful information for the source code of a node as a whole,
+    # but in Masamune we generally want the token itself, so we primarily get the token's location.
+    def token_location
+      location
+    end
+
+    # The source code of Prism nodes can be retrieved by calling #slice.
+    # However, the output tends to vary from node to node, and other methods
+    # like `name`, `message`, etc. are available in Prism nodes which means you
+    # have to know exactly which method to call the get the results you want.
+    #
+    # The `token` method simplifies all of this, and just gives you the
+    # variable, method name, etc. in String form by just calling `token`.
+    def token_value
+      content
+    end
+  end
+
+  class DefNode
+    node_predicate :method?
+    node_predicate :method_definition?
+    def token_location = name_loc
+    def token_value = name.to_s
+  end
+
+  class CallNode
+    def method? = true
```

Comment:
Okay, so we define the methods initially with `node_predicate`, and set the values `true` for all successive Nodes that match the same criteria. This works for me :+1:

Reply:
Yeah, it's just using inheritance to do it. So we get the default implementation for every type, which will always be `false`.